caps/eth.md

```diff
@@ -0,0 +1,223 @@
+helpful, nodes should inform peers of all blocks that they may not be aware of. Including
+hashes that the sending peer could reasonably be considered to know (due to the fact they
+were previously informed of because that node has itself advertised knowledge of the
+hashes through NewBlockHashes) is considered bad form, and may reduce the reputation of
+the sending node. Including hashes that the sending node later refuses to honour with a
+proceeding GetBlockHeaders message is considered bad form, and may reduce the reputation
+of the sending node.
+
+### Transactions (0x02)
+
+`[[nonce: P, receivingAddress: B_20, value: P, ...], ...]`
+
+Specify (a) transaction(s) that the peer should make sure is included on its transaction
+queue. The items in the list (following the first item `0x12`) are transactions in the
+format described in the main Ethereum specification. Nodes must not resend the same
+transaction to a peer in the same session. This packet must contain at least one (new)
+transaction.
+
+### GetBlockHeaders (0x03)
+
+`[block: {P, B_32}, maxHeaders: P, skip: P, reverse: P in {0, 1}]`
+
+Require peer to return a BlockHeaders message. Reply must contain a number of block
+headers, of rising number when `reverse` is `0`, falling when `1`, `skip` blocks apart,
+beginning at block `block` (denoted by either number or hash) in the canonical chain, and
+with at most `maxHeaders` items.
+
+### BlockHeaders (0x04)
+
+`[blockHeader_0, blockHeader_1, ...]`
+
+Reply to GetBlockHeaders. The items in the list (following the message ID) are block
+headers in the format described in the main Ethereum specification, previously asked for
+in a GetBlockHeaders message. This may validly contain no block headers if no block
+headers were able to be returned for the GetBlockHeaders query.
+
+### GetBlockBodies (0x05)
+
+`[hash_0: B_32, hash_1: B_32, ...]`
+
+Require peer to return a BlockBodies message. Specify the set of blocks that we're
+interested in with the hashes.
+
+### BlockBodies (0x06)
+
+`[[transactions_0, uncles_0] , ...]`
+
+Reply to GetBlockBodies. The items in the list are some of the blocks, minus the header,
+in the format described in the main Ethereum specification, previously asked for in a
+GetBlockBodies message. This may validly contain no items if no blocks were able to be
+returned for the GetBlockBodies query.
+
+### NewBlock (0x07)
+
+`[[blockHeader, transactionList, uncleList], totalDifficulty]`
+
+Specify a single block that the peer should know about. The composite item in the list
+(following the message ID) is a block in the format described in the main Ethereum
+specification.
+
+- `totalDifficulty` is the total difficulty of the block (aka score).
+
+### BlockHashesFromNumber (0x08)
+
+`[number: P, maxBlocks: P]`
+
+Requires peer to reply with a BlockHashes message. Message should contain block with
+that of number `number` on the canonical chain. Should also be followed by subsequent
```

Comment:
"Message should contain block with that of number..." not sure what that is supposed to mean.